packages/utils/src/lib/log-results.ts

```diff
@@ -0,0 +1,53 @@
+type PromiseFulfilledResult<T> = {
+  status: 'fulfilled';
+  value: T;
+};
+
+export function logMultipleResults<T>(
+  results: PromiseSettledResult<T>[],
+  messagePrefix: string,
+  succeededCallback?: (result: PromiseFulfilledResult<T>) => void,
+  failedCallback?: (result: PromiseRejectedResult) => void,
+) {
+  if (succeededCallback) {
+    const succeededResults = results.filter(
+      (result): result is PromiseFulfilledResult<T> =>
+        result.status === 'fulfilled',
+    );
+
+    logPluginExecution(
```

Comment:
What I don't understand is why is a specific function `logPluginExecution` inside a generic `logMultipleResults`?

We already have a this implemented correctly here:
- https://github.com/code-pushup/cli/blob/4c4fa0df08304dae4d3ab3edddbc38eb4fdd1848/packages/core/src/lib/implementation/persist.ts#L87
and used here:
- https://github.com/code-pushup/cli/blob/4c4fa0df08304dae4d3ab3edddbc38eb4fdd1848/packages/core/src/lib/collect-and-persist.ts#L20

In `logPluginExecution` we see that we use the generic function `logMultipleResults` and make a specific function out of it.

Reply:
Sorry 'bout that. Will it be fine if I change `logPluginExecution` to generic also?
I added this function basically becaise of the prev comment 'to encapsulate logging logic'.

Reply:
> What I don't understand is why is a specific function logPluginExecution inside a generic logMultipleResults?

Good point :+1: But I think that's just a naming problem. The implementation of `logPluginExecution` doesn't actually have anything to do with plugins in particular, so the name is misleading.

Reply:
Thanks @BioPhoton @matejchalk , renamed function to `logPromiseResults`, I believe it will help to avoid the confusion